app/app/src/main/kotlin/com/hedvig/app/feature/chat/viewmodel/ChatViewModel.kt

```diff
@@ -17,24 +19,42 @@ import giraffe.GifQuery
 import giraffe.UploadFileMutation
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.WhileSubscribed
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import kotlin.time.Duration.Companion.seconds
 
 class ChatViewModel(
   private val chatRepository: ChatRepository,
   private val chatClosedTracker: ChatEventStore,
   private val hAnalytics: HAnalytics,
+  private val featureManager: FeatureManager,
 ) : ViewModel() {
 
   private val retryChannel = RetryChannel()
 
   private val _messages = MutableStateFlow<ChatMessagesQuery.Data?>(null)
   val messages = _messages.asStateFlow()
 
+  /**
+   * [null] implies that it's still not evaluated. In those situations, not showing the chat at all is preferred.
+   */
+  val isChatDisabled: StateFlow<Boolean?> = flow {
```

Comment:
I think we still want to enable the chat if we cant evaluate the feature flag

Reply:
Yes, this is very important!
That's why the default value we've set in Unleash is "false".

What this will do is:
* Go here https://github.com/HedvigInsurance/android/blob/dbcb4ab8e582945584f730612c357e3204303117/app/hanalytics/hanalytics-core/src/main/kotlin/com/hedvig/android/hanalytics/HAnalyticsExperimentManager.kt#L24-L32
* If the experiments are not populated, ask the backend (at this point in the chat, this is basically never the case, we always have loaded experiments, but it'd work even if we did not)
* If it's in there, it will see what it was, otherwise, the generated code in HAnalytics looks like this:
```kotlin
suspend fun disableChat(): Boolean {
  try {
    val experiment = getExperiment("disable_chat")
    experimentEvaluated(experiment)

    return experiment.variant == "enabled"
  } catch (e: Exception) {
    experimentEvaluated(
      HAnalyticsExperiment(
        "disable_chat",
        "disabled",
      )
    )

    return false
  }
}
```
It will catch the exception if it was not in there, and default to `false`, aka the chat is *not* disabled.

So this should be fine because of this.

Reply:
But despite all this, at this point in the app, there's also no way that we've still not cached the experiments, so this should be a super fast evaluation in all cases.

Reply:
I see, sounds good!